src/Compilers/CSharp/Portable/Syntax/Syntax.xml

```diff
@@ -1776,46 +1776,41 @@
     </Field>
     <Field Name="Designation" Type="VariableDesignationSyntax"/>
   </Node>
-  <Node Name="DeconstructionPatternSyntax" Base="PatternSyntax">
-    <Kind Name="DeconstructionPattern" />
+  <Node Name="RecursivePatternSyntax" Base="PatternSyntax">
+    <Kind Name="RecursivePattern" />
     <Field Name="Type" Type="TypeSyntax" Optional="true" />
-    <Field Name="OpenParenToken" Type="SyntaxToken">
-      <Kind Name="OpenParenToken"/>
-    </Field>
-    <Field Name="SubPatterns" Type="SeparatedSyntaxList&lt;SubpatternElementSyntax&gt;"/>
-    <Field Name="CloseParenToken" Type="SyntaxToken">
-      <Kind Name="CloseParenToken"/>
-    </Field>
-    <Field Name="PropertySubpattern" Type="PropertySubpatternSyntax" Optional="true" />
+    <Field Name="DeconstructionPatternClause" Type="DeconstructionPatternClauseSyntax" Optional="true" />
+    <Field Name="PropertyPatternClause" Type="PropertyPatternClauseSyntax" Optional="true" />
     <Field Name="Designation" Type="VariableDesignationSyntax" Optional="true">
       <Kind Name="SingleVariableDesignation"/>
       <Kind Name="DiscardDesignation"/>
     </Field>
   </Node>
-  <Node Name="SubpatternElementSyntax" Base="CSharpSyntaxNode">
-    <Kind Name="SubpatternElement"/>
-    <Field Name="NameColon" Type="NameColonSyntax" Optional="true" />
-    <Field Name="Pattern" Type="PatternSyntax" />
-  </Node>
-  <Node Name="PropertyPatternSyntax" Base="PatternSyntax">
-    <Kind Name="PropertyPattern" />
-    <Field Name="Type" Type="TypeSyntax" Optional="true" />
-    <Field Name="PropertySubpattern" Type="PropertySubpatternSyntax"/>
-    <Field Name="Designation" Type="VariableDesignationSyntax" Optional="true">
-      <Kind Name="SingleVariableDesignation"/>
-      <Kind Name="DiscardDesignation"/>
+  <Node Name="DeconstructionPatternClauseSyntax" Base="CSharpSyntaxNode">
+    <Kind Name="DeconstructionPatternClause"/>
+    <Field Name="OpenParenToken" Type="SyntaxToken">
+      <Kind Name="OpenParenToken"/>
+    </Field>
+    <Field Name="Subpatterns" Type="SeparatedSyntaxList&lt;SubpatternElementSyntax&gt;"/>
```

Comment:
Just thinking outloud.  Instead of SubpatternElementSyntax.  It could PatternSyntax and PatternElementSyntax (or maybe PatternClauseSyntax).  Then, this would just be called .Elements or .Clauses.  "Sub" is a bit odd as it indicates this composes into a parent... but that's how all syntax works.  We never really say "Submember" or "Subanything".  It's somewhat of an implication of syntax in general

Reply:
@thoughts @gafter @jcouv ?

Reply:
But it isn't just a pattern. It includes a possible parameter, tuple element, field or property name. It can only occur as part of one of those two clauses.

---
In reply to: [187714189](https://github.com/dotnet/roslyn/pull/26782#discussion_r187714189) [](ancestors = 187714189)

Reply:
You may argue that the `id:1` in `{ id:1 }` is not an "argument", but neither is `id=1` in `Attr(id=1)`. So I think PatternArgument is fine.